### PR TITLE
#162953051: Allow missing items to be provided on room feedback

### PR DESCRIPTION
**Pull request by cwizard2011**
#### What does this PR do?
- Allow Users to add missing items to a room feedback

#### How should this be manually tested?
- Run the server
- Test the query at `localhost:5000/mrm`
- Run `createResponse` mutation with the parameters `room_id, question_id(for check question type)`

#### What are the relevant pivotal tracker stories?
[#162953051](https://www.pivotaltracker.com/story/show/162953051)

#### Screenshots
1. Response for Non-existence roomID
screen shot 2019-01-10 at 11 05 41 am

2. Response for non-existence resource in the room
screen shot 2019-01-10 at 11 06 17 am

3. Response for Missing item parameter omitted
screen shot 2019-01-10 at 11 07 05 am

4. Response for an empty list of missing items
screen shot 2019-01-10 at 11 06 49 am

5. Response for reported items
screen shot 2019-01-10 at 11 04 01 am

### CODE REVIEW

File: api/response/schema.py [hunk 1]
```diff
@@ -38,25 +42,48 @@ def mutate(self, info, **kwargs):
...
                                        room_id=kwargs['room_id'],
-                                       question_id=kwargs['question_id'])
+                                       question_id=kwargs['question_id'],
+                                       created_date=datetime.now())
                 rating.save()
                 return CreateResponse(response=rating)
         if question_type.lower() == 'check':
-            if 'check' not in kwargs:
-                raise GraphQLError("Provide a check response")
+            if 'missing_items' not in kwargs:
```
Comment (Bonifase) [inline api/response/schema.py]: Would it be appropriate to create a dynamic function to check the missing items in the kwargs and handle the validation, tell me what you think about this?

Comment (cwizard2011) [reply api/response/schema.py]: yes, I think that is a good idea, this will prevent the use of nested if statement. I will do that as soon as possible

Comment (cwizard2011) [reply api/response/schema.py]: @Bonifase I have implemented the change

**Comment by cwizard2011** [comment]
Okay

**Comment by Bonifase** [comment]
> @Bonifase I have implemented the required changes
Kindly have only one commit message per PR.

**Comment by cwizard2011** [comment]
Thanks, I have implemented the required changes @Bonifase
<https://github.com/Bonifase>

File: utilities/utility.py [hunk 2]
```diff
@@ -81,6 +81,16 @@ def validate_date_time_range(**kwargs):
         )
 
 
+def validate_missing_items_field(**kwargs):
+    """
+    Function to validate the missing item field
+    when saving a check question response
+    :params kwargs
+    """
+    if 'missing_items' not in kwargs:
```
Comment (Bonifase) [inline utilities/utility.py]: Would it be possible to be specific in your validation instead of generalizing items?

**Comment by cwizard2011** [comment]
Yes

**Pull request merged**